src/main/java/frc/robot/subsystems/swerve/SwerveCommandsBuilder.java

```diff
@@ -0,0 +1,173 @@
+import frc.utils.auto.PathPlannerUtils;
+import frc.utils.calibration.swervecalibration.WheelRadiusCharacterization;
+import frc.utils.calibration.sysid.SysIdCalibrator;
+import frc.utils.utilcommands.InitExecuteCommand;
+
+import java.util.Set;
+import java.util.function.DoubleSupplier;
+import java.util.function.Supplier;
+
+public class SwerveCommandsBuilder {
+
+	private final Swerve swerve;
+	private final Modules modules;
+	private final SysIdCalibrator steerCalibrator;
+	private final SysIdCalibrator driveCalibrator;
+
+	public SwerveCommandsBuilder(Swerve swerve) {
+		this.swerve = swerve;
+		this.modules = swerve.getModules();
+		this.steerCalibrator = new SysIdCalibrator(
+			modules.getModule(ModuleUtils.ModulePosition.FRONT_LEFT).getSteerSysIdConfigInfo(),
+			swerve,
+			modules.getModule(ModuleUtils.ModulePosition.FRONT_LEFT)::setSteerVoltage
+		);
+		this.driveCalibrator = new SysIdCalibrator(
+			modules.getModule(ModuleUtils.ModulePosition.FRONT_LEFT).getDriveSysIdConfigInfo(),
+			swerve,
+			modules::setDrivesVoltage
+		);
+	}
+
+	public Command steerCalibration(boolean isQuasistatic, SysIdRoutine.Direction direction) {
+		return steerCalibrator.getSysIdCommand(isQuasistatic, direction).withName("Steer calibration");
+	}
+
+	public Command driveCalibration(boolean isQuasistatic, SysIdRoutine.Direction direction) {
+		Command sysIdCommand = driveCalibrator.getSysIdCommand(isQuasistatic, direction);
+		sysIdCommand.getRequirements().clear();
+
+		return new SequentialCommandGroup(
+			pointWheels(new Rotation2d(), false).until(
+				() -> modules.isSteerAtTargetPositions(
+					SwerveConstants.CALIBRATION_MODULE_ANGLE_TOLERANCE,
+					SwerveConstants.CALIBRATION_MODULE_ANGULAR_VELOCITY_PER_SECOND_DEADBAND
+				)
+			),
+			new ParallelDeadlineGroup(sysIdCommand, pointWheels(new Rotation2d(), false))
+		).withName("Drive calibration");
+	}
+
+
+	public Command pointWheels(Rotation2d targetSteerPosition, boolean optimize) {
+		return new RunCommand(() -> modules.pointWheels(targetSteerPosition, optimize), swerve)
+			.withName("Point wheels to: " + targetSteerPosition);
+	}
+
+	public Command pointWheelsInCircle() {
+		return new RunCommand(modules::pointWheelsInCircle, swerve).withName("Point wheels in circle");
+	}
+
+	public Command pointWheelsInX() {
+		return new RunCommand(modules::pointWheelsInX, swerve).withName("Point wheels in X");
+	}
+
+
+	public Command setTargetModuleStates(Supplier<SwerveModuleState[]> statesSupplier, boolean isClosedLoop) {
+		return new RunCommand(() -> modules.setTargetStates(statesSupplier.get(), isClosedLoop), swerve).withName("Set states by " + "supplier");
+	}
+
+
+	public Command wheelRadiusCalibration() {
+		return new SequentialCommandGroup(
+			pointWheelsInCircle().until(
+				() -> swerve.getModules()
+					.isSteerAtTargetPositions(
+						SwerveConstants.CALIBRATION_MODULE_ANGLE_TOLERANCE,
+						SwerveConstants.CALIBRATION_MODULE_ANGULAR_VELOCITY_PER_SECOND_DEADBAND
+					)
+			),
```

Comment:
This command can be created in a function, instead of code duplication

- [ ] create command in method and call it everywhere you need it